### DISCUSSION TITLE
[data-importer] Tries to connect to https://app:8080/ (curl error 35)

### DISCUSSION THREAD

**Discussion by laikq** [firefly-iii/firefly-iii#7004] [Firefly III] [answered]
Hi,

I'm using the data importer with docker compose as described in the documentation. I've set `FIREFLY_III_URL=http://app:8080`, and `VANITY_URL=https://<my-firefly-install>.de`. Both firefly and the data importer run behind a reverse proxy so that I can access them via HTTPS. The reverse proxy also forces HTTPS by redirecting HTTP to HTTPS.

When I access `import.<my-firefly-install>.de`, I get the error

```
cURL error 35: error:1408F10B:SSL routines:ssl3_get_record:wrong version number (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://app:8080/api/v1/about
```

(The webpage loads fine, it says there was a connection error to firefly.)

It seems to me that the data importer tries to access Firefly via HTTPS. But this happens inside of the Docker network, so why should it try HTTPS (when `FIREFLY_III_URL` also does not use HTTPS?). I feel like I'm missing something, thanks for any help!

**Comment by JC5** [comment, marked as answer]
If this is done by your reverse proxy, it's access logs should show the data importer connecting to the webserver.

**Reply by laikq** [reply]
Thanks for your answer! I looked into the log of nginx, but it does not mention a failed SSL attempt / wrong version number. I'm pretty sure this `curl` attempt doesn't go through the reverse proxy, it is just inter-container communication. It *should* use HTTP (but doesn't).

I can also reproduce this by running `docker exec -it $FIDI_CONTAINER_ID curl https://app:8080/api/v1/about`: This gets me exactly the error I also get on the webpage (`wrong version number`). Whereas when I run `docker exec -it $FIDI_CONTAINER_ID curl http://app:8080/api/v1/about` (https → http), it gives me a valid html answer (redirect to `/login`).

(stdout of the firefly app logs an entry `"\x16\x03\x01\x02" 400 0 "-" "-"` when I access the data importer. I presume this is the https attempt which firefly of course can't understand.)

**Reply by JC5** [reply]
You access the data importer in the browser, right? Try browsing to `/flush`, see if that helps.

**Reply by laikq** [reply]
Wait, this is weird: I tested this on a different machine and there it works, I can import files. But on the machine where I tried this originally it still doesn't work (even after disabling all extensions).

**Reply by JC5** [reply]
The Data Importer has no storage, so it saves some values in a session cookie. Browse to `/flush` or clear your cookies, that should do it.

**Reply by laikq** [reply]
Indeed, after clearing cookies I now get a different error (invalid_client), but I think this is really a different topic (I changed authentication from token to client_id so that not everyone can upload data). Thanks for all the help! I will try to solve this by myself from here on :) [Edit: Fixed it, invalid_client was caused by me using a wrong CLIENT_ID.]

TL;DR: Browse to `/flush` and clear cookies.
